pkg/publicshare/manager/json/json.go

```diff
@@ -343,10 +343,22 @@ func (m *manager) UpdatePublicShare(ctx context.Context, u *user.User, req *link
 		old, _ := json.Marshal(share.Permissions)
 		new, _ := json.Marshal(req.Update.GetGrant().Permissions)
 
-		if m.writeableShareMustHavePassword && publicshare.IsWriteable(req.GetUpdate().GetGrant().GetPermissions()) && !share.PasswordProtected {
+		if m.writeableShareMustHavePassword &&
+			publicshare.IsWriteable(req.GetUpdate().GetGrant().GetPermissions()) &&
+			(!share.PasswordProtected && req.GetUpdate().GetGrant().GetPassword() == "") {
 			return nil, publicshare.ErrShareNeedsPassword
 		}
 
+		if req.GetUpdate().GetGrant().GetPassword() != "" {
```

Comment:
Why do we change permissions and the password together? Shouldn't a password change be done within the `link.UpdatePublicShareRequest_Update_TYPE_PASSWORD` case?